src/ui/components/Library/Team/View/TestRuns/hooks/useTestRuns.ts

```diff
@@ -1,25 +1,43 @@
-import { useContext } from "react";
-import { useImperativeCacheValue } from "suspense";
+import { useContext, useEffect, useMemo, useState } from "react";
+import { STATUS_PENDING, Status, useImperativeIntervalCacheValues } from "suspense";
 
 import { GraphQLClientContext } from "replay-next/src/contexts/GraphQLClientContext";
 import { TestRun } from "shared/test-suites/TestRun";
 import { TeamContext } from "ui/components/Library/Team/TeamContextRoot";
-import { testRunsCache } from "ui/components/Library/Team/View/TestRuns/suspense/TestRunsCache";
 import useToken from "ui/utils/useToken";
 
+import { TimeFilterContext } from "../../TimeFilterContextRoot";
+import { testRunsIntervalCache } from "../suspense/TestRunsCache";
+
 const EMPTY_ARRAY: any[] = [];
 
-export function useTestRuns(): TestRun[] {
+export function useTestRuns(): { testRuns: TestRun[]; status: Status } {
   const graphQLClient = useContext(GraphQLClientContext);
   const { teamId } = useContext(TeamContext);
+  const { startTime, endTime } = useContext(TimeFilterContext);
+  const [testRuns, setTestRuns] = useState<TestRun[]>([]);
 
   const accessToken = useToken();
 
-  const { value = EMPTY_ARRAY } = useImperativeCacheValue(
-    testRunsCache,
+  const { value = EMPTY_ARRAY, status } = useImperativeIntervalCacheValues(
+    testRunsIntervalCache,
+    startTime.getTime(),
+    endTime.getTime(),
     graphQLClient,
     accessToken?.token ?? null,
     teamId
   );
-  return value;
+
+  useEffect(() => {
+    if (status === STATUS_PENDING) {
+      // To avoid resetting the test runs to an empty array when loading
+      return;
+    }
+
+    setTestRuns(value);
+  }, [value, status]);
```

Comment:
It's being used to retain the data from previous date range while the next range is loading. Is there another way we should be achieving that?

Reply:
I see. I think this is because we're not using Suspense, so we don't get that (the is-pending, dimmed behavior) for free.

Can we fetch this data using suspense (e.g. cache.read) rather than the imperative hook?

I don't know if there's any gotchas in this, but it seems like it would be nice to remove `testRuns` from the context , and just load it inside of `TestRunsList` or `TestRunDetailsPanel` using Suspense like we do with the other data (aka rename this hook to `useTestRunsSuspends`)

Reply:
Cool 👍🏼  I'm having a bit of trouble keeping up with what's the current state here given the size of the PR

Reply:
No worries. I was waiting to tag you until it had landed but you got ahead of me :)